api/nodes/views.py

```diff
@@ -1390,8 +1390,6 @@ def get_object(self):
             self.kwargs[node_link_lookup_url_kwarg],
             'node link'
         )
-        # May raise a permission denied
-        self.check_object_permissions(self.request, node_link)
```

Comment:
Okay, here's my problem with this. I modified your permissions classes so that they were:

```py
    permission_classes = (
        FailPermissionCheck,
        ContributorOrPublicForPointers,
        base_permissions.TokenHasScope,
        drf_permissions.IsAuthenticatedOrReadOnly,
        ExcludeWithdrawals
    )
```
Note the one at the top. It is 
```py
class FailPermissionCheck(BasePermission):

    def has_object_permission(self, request, view, obj):
        return False
```

Which, if I did it right, means it should never get anything but a 403 no matter what. But it doesn't, because you removed the check_object_permissions for this view. 

Most or all of the stuff you did below should be done by the permissions class. Take a look at how `ContributorOrPublicForPointers` works; it loads the node object and does checks on that.